onnxruntime/contrib_ops/cpu/transformers/beam_search.cc

```diff
@@ -151,33 +151,28 @@ Status BeamSearch::SetupSubgraphExecutionInfo(const SessionState& session_state,
     }
   } else if (parameters_.model_type == IGenerationParameters::kModelTypeWhisper) {
     if (attribute_name == "encoder") {
-      ORT_ENFORCE(t5_encoder_subgraph_ == nullptr,
+      ORT_ENFORCE(whisper_encoder_subgraph_ == nullptr,
                   "SetupSubgraphExecutionInfo should only be called once for each subgraph.");
-      t5_encoder_subgraph_ = std::make_unique<WhisperEncoderSubgraph>(node,
-                                                                      attribute_name,
-                                                                      subgraph_session_state.GetGraphViewer());
-      ORT_RETURN_IF_ERROR(t5_encoder_subgraph_->Setup(session_state, subgraph_session_state));
-      encoder_feeds_fetches_manager_ = t5_encoder_subgraph_->GetFeedsFetchesManager();
-
-      if (parameters_.decoder_start_token_id < 0) {
-        ORT_RETURN_IF(t5_encoder_subgraph_->num_subgraph_inputs != 2,
-                      "Encoder subgraph shall have 2 inputs when decoder_start_token_id attribute is empty");
-      } else {
-        ORT_RETURN_IF(t5_encoder_subgraph_->num_subgraph_inputs != 3,
-                      "Encoder subgraph shall have 3 inputs when decoder_start_token_id attribute is available");
-      }
+      whisper_encoder_subgraph_ = std::make_unique<WhisperEncoderSubgraph>(node,
+                                                                           attribute_name,
+                                                                           subgraph_session_state.GetGraphViewer());
+      ORT_RETURN_IF_ERROR(whisper_encoder_subgraph_->Setup(session_state, subgraph_session_state));
+      encoder_feeds_fetches_manager_ = whisper_encoder_subgraph_->GetFeedsFetchesManager();
+
+      ORT_RETURN_IF(whisper_encoder_subgraph_->num_subgraph_inputs != 2,
+                    "Encoder subgraph shall have 3 inputs (encoder_input_ids, decoder_input_ids)");
```

Comment:
Message should read `shall have 2 inputs`

Reply:
Fixed